datasets/ua/war_sanctions/crawler.py

```diff
@@ -0,0 +1,397 @@
+    data = {}
+    for row in details_container.findall(".//div[@class='row']"):
+        label_elem = row.find(".//div[@class='col-12 col-sm-4 yellow']")
+        value_elem = row.find(".//div[@class='col-12 col-sm-8']")
+
+        if label_elem is not None and value_elem is not None:
+            data = extract_label_value_pair(label_elem, value_elem, data=data)
+
+    name = data.pop("Name")
+    name_abbr = data.pop("Abbreviated name of the legal entity", None)
+    reg_num = data.pop("Registration number")
+
+    company = context.make("Company")
+    company.id = context.make_id(name, name_abbr, reg_num)
+    company.add("name", name)
+    company.add("name", name_abbr)
+    company.add("registrationNumber", reg_num)
+    company.add("address", data.pop("Address"))
+    company.add("country", data.pop("Country"))
+    company.add("taxNumber", data.pop("Tax Number"))
+    company.add("sourceUrl", data.pop("Links").split(" | "))
+    archive_links = data.pop("Archive links", None)
+    if archive_links is not None:
+        for archive_link in archive_links.split(" | "):
+            company.add("sourceUrl", archive_link)
+
+    company.add("topics", "poi")
+    sanction = h.make_sanction(context, company)
+    sanction.add("reason", data.pop("Reasons"))
+    sanction.add("sourceUrl", link)
+
+    context.emit(company, target=True)
+    context.emit(sanction)
+    context.audit_data(data)
+
+
+def extract_next_page_url(doc, base_url, next_xpath):
+    doc.make_links_absolute(base_url)
+    # next page <a> element extraction using xpath
+    next_link_element = doc.xpath(next_xpath)
+
+    if next_link_element:
+        next_link = next_link_element[0]
+        return next_link.get("href")
+
+    return None
+
+
+def crawl(context):
+    main_page = context.fetch_html(context.data_url, cache_days=3)
+    node = main_page.find(
+        ".//section[@class='sections d-flex flex-wrap align-items-stretch justify-content-center mb-5 pl-5 pr-5 medium']"
+    )
+    h.assert_dom_hash(node, "dbb9a924d940b3a69a132a102e04dcf0f9fbfc5e")
+
+    for link_info in LINKS:
+        base_url = link_info["url"]
+        data_type = link_info["type"]
+        current_url = base_url
+        visited_pages = 0
+        while current_url and visited_pages < 4:  # emergency exit check
+            doc = context.fetch_html(current_url)
+            if doc is None:
+                print(f"Failed to fetch {current_url}")
+                break
+            context.log.info(f"Processing {current_url}")
+            crawl_index_page(context, current_url, data_type)
+
+            # xpath to find the next page link
+            next_xpath = "//ul[@class='pagination']//li[@class='next']/a"
+
+            # get the next page URL, if exists
+            next_url = extract_next_page_url(doc, base_url, next_xpath)
+
+            if next_url:
+                current_url = next_url
+                visited_pages += 1
+            else:
+                break
```

Comment:
Nice, you're already checking if current_url is set, so you can trim this down

```suggestion
            next_url = extract_next_page_url(doc, base_url, next_xpath)
            current_url = next_url
            visited_pages += 1
```